ports/espressif/common-hal/mdns/Server.c

```diff
@@ -37,6 +37,7 @@ STATIC bool inited = false;
 
 void mdns_server_construct(mdns_server_obj_t *self, bool workflow) {
     if (inited) {
+        self->inited = false;
```

Comment:
Clearing the `inited` flag in construct seems weird. If this is correct, can you add a comment?

Reply:
Should I return a bool instead? That was my other option. The web workflow is a bit of a weird world where we have static memory and having it deinit on failure is convenient for later use.

Reply:
Since I don't understand what it's doing, I can't really say how I'd like you to fix it.  It just feels weird that "construct" would ever set a flag called "inited" to False and I'm trying to understand what's going on.

What does the "self->inited" flag _mean_? Is there a better name for it, or a comment that could explain it?

Since it tests out good, probably a different name or a comment explaining what's going on would be helpful, rather than changing the flow of the code.